cpu/sam0_common/periph/spi.c

```diff
@@ -95,6 +92,22 @@ void spi_init_pins(spi_t bus)
     gpio_init_mux(spi_config[bus].miso_pin, spi_config[bus].miso_mux);
     gpio_init_mux(spi_config[bus].mosi_pin, spi_config[bus].mosi_mux);
     gpio_init_mux(spi_config[bus].clk_pin, spi_config[bus].clk_mux);
+
+    /* wake up device */
+    poweron(bus);
```

Comment:
Powering on the bus should go above `dev(bus)->CTRLA.reg |= SERCOM_SPI_CTRLA_SWRST;` 
Otherwise we might end up stuck in while` ((dev(bus)->CTRLA.reg & SERCOM_SPI_CTRLA_SWRST)`

Reply:
But `spi_init_pins()` is called before `dev(bus)->CTRLA.reg |= SERCOM_SPI_CTRLA_SWRST;`

Reply:
Oh right sorry for the noise.
Github rendering tricks me...